apps/site/src/pages/settings/billing/upgrade/change-payment-method-modal.tsx

```diff
@@ -0,0 +1,287 @@
+                        sx={({ palette, spacing }) => ({
+                          padding: spacing(0.5, 0.75),
+                          color: palette.purple[700],
+                          background: palette.purple[20],
+                          borderRadius: 1,
+                          textTransform: "uppercase",
+                          fontSize: 11,
+                        })}
+                      >
+                        <FontAwesomeIcon
+                          icon={faCheck}
+                          sx={{ marginRight: 0.75, fontSize: 12 }}
+                        />
+                        <strong>Chosen</strong>
+                      </Box>
+                    </Fade>
+                    {isChosen ? null : (
+                      <Box
+                        component="span"
+                        className="switch-text"
+                        sx={{
+                          position: "absolute",
+                          left: 0,
+                          top: 3,
+                          textTransform: "uppercase",
+                          fontSize: 11,
+                          fontWeight: 600,
+                          opacity: 0,
+                          transition: ({ transitions }) =>
+                            transitions.create("opacity"),
+                        }}
+                      >
+                        Switch
+                        <FontAwesomeIcon
+                          icon={faArrowRight}
+                          sx={{
+                            marginLeft: 0.75,
+                            fontSize: 12,
+                            position: "relative",
+                            top: -1,
+                          }}
+                        />
+                      </Box>
+                    )}
+                  </Box>
+                }
+                endAdornment={
+                  <Link
+                    href={{
+                      pathname: cardDetailsPanelPageAsPath,
+                      query: { id: paymentMethod.id },
+                    }}
+                  >
+                    <Tooltip title="Change card details">
+                      <IconButton
+                        onClick={(event) => {
+                          event.stopPropagation();
+                        }}
+                        sx={{
+                          "& svg": {
+                            color: ({ palette }) => palette.gray[40],
+                          },
+                        }}
+                      >
+                        <FontAwesomeIcon
+                          icon={faEdit}
+                          sx={{ marginRight: 0.75, fontSize: 12 }}
+                        />
+                      </IconButton>
+                    </Tooltip>
+                  </Link>
+                }
+              />
+            </Button>
+          );
+        })}
+        <Box marginBottom={4}>
+          <Collapse in={addingPaymentMethod}>
```

Comment:
Non-blocking: I think this animation could look better
![new](https://user-images.githubusercontent.com/37453800/217606745-d096f80d-c3f3-4bdc-8a70-f887b08ed2c3.gif)

LMK if you want to change it and I'll send you the code diff.

Reply:
Sure, if you prefer it feel free to send it my way, or use "add a suggestion" button in a separate GH comment

Reply:
Sent via slack

Reply:
Added in https://github.com/blockprotocol/blockprotocol/pull/945/commits/021535184d1629bbf2802d155daffcb1b4c0e5b3